Source/WebCore/platform/cocoa/PlatformSpeechSynthesizerCocoa.mm

```diff
@@ -90,9 +90,26 @@ - (WebSpeechSynthesisWrapper *)initWithSpeechSynthesizer:(WebCore::PlatformSpeec
         return nil;
 
     m_synthesizerObject = synthesizer;
+#if HAVE(AVSPEECHSYNTHESIS_VOICES_CHANGE_NOTIFICATION)
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(availableVoicesDidChange:) name:AVSpeechSynthesisAvailableVoicesDidChangeNotification object:nil];
+#endif
     return self;
 }
 
+- (void)dealloc
+{
+#if HAVE(AVSPEECHSYNTHESIS_VOICES_CHANGE_NOTIFICATION)
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:AVSpeechSynthesisAvailableVoicesDidChangeNotification object:nil];
+#endif
+    [super dealloc];
+}
+
+- (void)availableVoicesDidChange
```

Comment:
this method should have a parameter of NSNotification on it

when you register it has the colon

selector:@selector(availableVoicesDidChange:)

Reply:
Will change.

Reply:
Wonder if we need to wrap this whole method inside 

f HAVE(AVSPEECHSYNTHESIS_VOICES_CHANGE_NOTIFICATION)